src/components/App/AppSagas.js

```diff
@@ -0,0 +1,33 @@
+import { put } from 'redux-saga/dist/redux-saga-effects-npm-proxy.cjs';
+
+import auth from 'auth';
+import { CHANGE_PAGE_ACTION } from 'components/App/AppActions';
+import { PAGES_ENUM } from 'components/App/AppConstants';
+import thoughtStorage from 'storage/thought-storage';
+import googleDriveApi from 'api/google-drive-api';
+import googleApiLoader from 'api/google-api-loader';
+import { hideSpinner, showSpinner } from 'components/Spinner/SpinnerSagas';
+
+export function setPageAction(data) {
+  return put({ type: CHANGE_PAGE_ACTION, data });
+}
+
+export function* loadApp() {
+  console.info('Loading app...');
+
+  yield showSpinner('Loading Google Api');
+  yield googleApiLoader.load();
+  yield googleDriveApi.loadDriveApi();
+  yield thoughtStorage.scanDrive();
+  yield setPageAction(PAGES_ENUM.NOTES);
```

Comment:
yields look cool :) so better then promises.

Where are the errors are going to be handled though?

Reply:
Sagas using try/catch construction for this. And as a result,  ACTION_NAME_FAILED will be dispatched. 
https://redux-saga.js.org/docs/basics/ErrorHandling.html